### DISCUSSION TITLE
How to properly change one subscription with another?

### DISCUSSION THREAD

**Discussion by Fayozjon** [bpuig/laravel-subby#114] [Q&A]
Hello! 
As i wrote at #109  i done same but now my NEW subscription PLAN
has OLD ONES date of issue (closing time) = ENDS_AT

**Comment by Fayozjon** [comment]
``` 

public function activatePlan($id){
        $plan = Plan::findOrFail($id);
        $subscription = auth()->user()->subscription();
        if(null == $subscription){
            auth()->user()->newSubscription($plan);
            $subscription = auth()->user()->subscription();
        }
        $subscription->changePlan($plan);

        $subscription->syncPlan(null, true, true);
        return back()->with('message',__("Tariff changed to ".$plan->name));
    }
    
    
```

**Reply by bpuig** [reply]
Hello!

It is not necessary to do `syncPlan`, since `changePlan` already does it, but you need to `renew`. Let me give you an example of the function with this base imaginary data:

`$plan1` is a 1 month plan.
`$plan2` is a 6 month plan.

$user is currently on `$plan1` and subscribed on January 1st. And now we are on January 22nd.
And you have an imaginary payment logic that we will call `doPayment`;

```php
// Pay the new plan
$user->doPayment($plan2->price, $plan2->currency); 

// Change to the new plan, by default usage is reset and on February 1st,
// the new period will be until August 1st. 
// But right now current period still stays January 1st to February 1st
$user->subscription()->changePlan($plan2); 

// Since you want the changes to be inmediately applied because it has been
// paid in full, you renew your subscription period so now it will end on July 22nd (6 months from now)
$user->subscription()->renew(); 
```
Why is that "complicated"? Because you may want to not renew, leave the usage and prorate price. Or any kind of combination, I try to keep the functions as open and combinable as possible.

Hope it helps

**Reply by Fayozjon** [reply]
Thank you @bpuig  for you patiens and anwsers ) yes it helped
